pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -517,15 +533,18 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, nodeName string) (diskID string
 		return "", "", err
 	}
 
-	// find SCSI controller to attach the disk
-	var newSCSICreated bool = false
+	var diskControllerType = vs.cfg.Disk.SCSIControllerType
+	// find SCSI controller of particular type from VM devices
+	var diskController = getSCSIController(vmDevices, diskControllerType)
+
+	var newSCSICreated = false
 	var newSCSIController types.BaseVirtualDevice
-	diskController, err := vmDevices.FindDiskController(vs.cfg.Disk.DiskController)
-	if err != nil {
-		// create a scsi controller if there is not one
-		newSCSIController, err := vmDevices.CreateSCSIController(vs.cfg.Disk.SCSIControllerType)
+	// creating a scsi controller as there is none found of controller type defined
+	if diskController == nil {
+		glog.V(4).Infof("Creating a SCSI controller of %v type", diskControllerType)
+		newSCSIController, err := vmDevices.CreateSCSIController(diskControllerType)
 		if err != nil {
-			glog.V(3).Infof("Cannot create new SCSI controller - %v", err)
+			glog.Errorf("cannot create new SCSI controller - %v", err)
 			return "", "", err
```

Comment:
The context for the error (`Cannot create new SCSI controller`) needs to be returned in the error.